include/libtorrent/session.hpp

```diff
@@ -131,6 +121,44 @@ namespace libtorrent
 		boost::shared_ptr<aux::session_impl> m_impl;
 	};
 
+	struct TORRENT_EXPORT session_params
```

Comment:
I would imagine this class would be a lot more useful if you could construct it, and add stuff to it in a chain. Something like: `session s(session_params(settings, flags).add_extension(foo).add_extension(bar));`

I'm not sure that's possible though